backend/cdb/api/db/graphql/socio_pro.py

```diff
@@ -1,114 +1,67 @@
-
-    def gql_variables(self):
-        return super().gql_variables() | {
-            "id": str(self.id),
-        }
-
-
-class SituationsToAdd(BaseModel):
-    notebookId: UUID
-    situationId: UUID
-
-    def gql_variables(self):
-        return {
-            "notebookId": str(self.notebookId),
-            "situationId": str(self.situationId),
-        }
 
+def get_professional_project_to_add_gql_variables(professional_project):
+    return get_professional_project_common_gql_variables(professional_project) | {
+        "notebookId": str(professional_project.notebookId),
+    }
 
-class Input(BaseModel):
-    educationLevel: str | None
-    id: UUID
-    lastJobEndedAt: str | None
-    professionalProjectIdsToDelete: List[UUID]
-    professionalProjectsToAdd: List[ProfessionalProjectToAdd]
-    professionalProjectsToUpdate: List[ProfessionalProjectToUpdate]
-    rightRqth: bool
-    situationIdsToDelete: List[UUID]
-    situationsToAdd: List[SituationsToAdd]
-    workSituation: str | None
-    workSituationDate: str | None
-    workSituationEndDate: str | None
 
+def get_situation_to_add_gql_variables(situation_to_add):
+    return {
+        "notebookId": str(situation_to_add.notebookId),
+        "situationId": str(situation_to_add.situationId),
+    }
 
-class UpdateSocioProActionPayload(BaseModel):
-    action: Action
-    input: Input
-    request_query: str
-    session_variables: Dict[str, str]
 
-    def gql_variables(self):
-        return {
-            "id": str(self.input.id),
-            "workSituation": self.input.workSituation,
-            "workSituationDate": self.input.workSituationDate,
-            "workSituationEndDate": self.input.workSituationEndDate,
-            "rightRqth": self.input.rightRqth,
-            "educationLevel": self.input.educationLevel,
-            "lastJobEndedAt": self.input.lastJobEndedAt,
-            "professionalProjectIdsToDelete": [
-                str(id) for id in self.input.professionalProjectIdsToDelete
-            ],
-            "professionalProjectsToAdd": [
-                p.gql_variables() for p in self.input.professionalProjectsToAdd
-            ],
-            "professionalProjectsToUpdate": [
-                {
-                    "where": {
-                        "id": {"_eq": p.id},
-                    },
-                    "_set": {
-                        "mobilityRadius": p.mobilityRadius,
-                        "romeCodeId": p.romeCodeId,
-                        "contractTypeId": p.contractTypeId,
-                        "employmentTypeId": p.employmentTypeId,
-                        "hourlyRate": p.hourlyRate,
-                    },
-                }
-                for p in self.input.professionalProjectsToUpdate
-            ],
-            "situationsToAdd": [s.gql_variables() for s in self.input.situationsToAdd],
-            "situationIdsToDelete": [str(id) for id in self.input.situationIdsToDelete],
-        }
+def gql_variables(input):
```

Comment:
On pourrait rajouter le type 
```suggestion
def gql_variables(input: Input):
```